Sources/MongoSwift/MongoCollection.swift

```diff
@@ -103,6 +122,41 @@ public struct MongoCollection<T: Codable> {
         return self._client.operationExecutor.execute(operation, client: self._client, session: session)
     }
 
+    /**
+     * Renames this collection with the specified options.
```

Comment:
I think we need to include a bit more detail here to cover some of the confusion users might have with this method. Specifically, we need to emphasize that this method renames the collection server side and returns a handle to the renamed collection. The current handle (which the method is invoked on) will continue to refer to the old collection, which will now be empty.

Another important point to mention is that this method does not work on sharded collections.